R/AAA.R

```diff
@@ -54,7 +63,26 @@ load_stuff <- function() {
       assign("evolution_status", unname(evolution_status), envir=.spOptions)
     }
   }
-  Smess <- paste("The legacy packages maptools, rgdal, and rgeos, underpinning this package\nwill retire shortly. Please refer to R-spatial evolution reports on\nhttps://r-spatial.org/r/2023/05/15/evolution4.html for details.\nThis package is now running under evolution status", get_evolution_status(), "\n")
+  startup_message <- unlist(options("sp_startup_message"))
+  if (!is.null(startup_message)) {
+    startup_message <- as.character(startup_message)
+    stopifnot(is.character(startup_message))
+    stopifnot(isTRUE(startup_message %in% c("attach", "load", "none")))
+    assign("startup_message", unname(startup_message), envir=.spOptions)
+  } else {
+    startup_message <- Sys.getenv("_SP_STARTUP_MESSAGE_")
+    if (nchar(startup_message) > 0L) {
+      startup_message <- as.character(startup_message)
+      stopifnot(is.character(startup_message))
+      stopifnot(isTRUE(startup_message %in% c("attach", "load", "none")))
+      assign("startup_message", unname(startup_message), envir=.spOptions)
+    }
+  }
+}
+
+smess_func <- function() {
+  where <- get("startup_message", envir=.spOptions)
+  Smess <- paste("The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,\nwhich was just ", where, "ed, will retire in October 2023.\nPlease refer to R-spatial evolution reports for details, especially\nhttps://r-spatial.org/r/2023/05/15/evolution4.html.\nThe sp package is now running under evolution status ", get_evolution_status(), "\n", ifelse(get_evolution_status() == 2L, "     (status 2 uses the sf package in place of rgdal)\n", ""), sep="")
```

Comment:
Thanks, this is an important change in wording. As from sp 1.6-1, loading any package that imports sp outputs a verbose and confusing message about "this package" where sp is meant, e.g.:
```
R> library(rmapshaper)
The legacy packages maptools, rgdal, and rgeos, underpinning this package
will retire shortly. Please refer to R-spatial evolution reports on
https://r-spatial.org/r/2023/05/15/evolution4.html for details.
This package is now running under evolution status 2 
```

Reply:
The current status in this PR is:
```
> loadNamespace("sp")
The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,
which was just loaded, will retire in October 2023.
Please refer to R-spatial evolution reports for details, especially
https://r-spatial.org/r/2023/05/15/evolution4.html.
It may be desirable to make the sf package available;
package maintainers should consider adding sf to Suggests:.
The sp package is now running under evolution status 2
     (status 2 uses the sf package in place of rgdal)
```
Is this better? See also https://github.com/Trackage/trip/issues/52#issuecomment-1572774913 for @mdsumner 's helpful nudge. There are now also an environment variable and an option to change the default setting. From reactions on twitter, some ecologists see the end of the world coming from retiring packages breaking legacy workflows, hence the noisy start-up message, telling them where to go for information.

Reply:
The verbose start-up message is certainly useful for when sp is being attached. I am less sure about loading as I guess many sp-importing packages have already adapted to evolution status 2 in which case the message is mostly redundant and rather disturbing (but I guess you want to spread the word).

Reply:
Current rev-deps show about 220 packages failing when retiring packages are not on the library path and `_R_CHECK_SUGGESTS_ONLY_=true _R_CHECK_FORCE_SUGGESTS_=FALSE`. Many of these have stale declarations from when `raster` needed `rgdal` and `rgeos` in Suggests:, most of these were alerted in December 2022, but haven't responded. Hence the nneed for noise at this point. 

The evidential background includes the GDAL barn-raising process, when the representation of coordinate reference systems was changed upstream of us - I see plenty of package code turning off messages, and staying with explicitly defunct representations. So lots of hheads buried securely in sand, I'm afraid.

Reply:
Thanks, I see. The obvious downside of the "load" default for "sp_startup_message" is that the loading of all sp-importing packages that have already been updated or are otherwise unaffected by the evolution becomes unnecessarily noisy. So I think in the near future, sp should really only speak out when attached.

Reply:
Exactly. The aim is to track the updating/mitigation rate, and relax the message to default "attach" as soon as there is realistic (for some definition) movement. I think few users `update.packages` at all frequently, so may well still be on `sp` 1.6-0 at best. I'm thinking of starting a cron job to pull nightly and CMD check new versions of the 200+ packages on my watch list as they update - any ideas?

Reply:
Watchlist check running, reporting thrice weekly at: https://github.com/r-spatial/evolution/tree/main/watchlist_output. The input were packages with `CMD check` errors or warnings with `_SP_EVOLUTION_STATUS_=2 R_LIBS=/home/rsb/lib/r_libs _R_CHECK_SUGGESTS_ONLY_=true _R_CHECK_FORCE_SUGGESTS_=FALSE` (no retiring packages on library path), initially 223 of 696 packages `package_dependencies(packages = c("sp", "maptools", "rgdal", "rgeos"), pdb, which = "most", recursive = FALSE, reverse = TRUE)` reduced to 222 by dropping `rgeos`, the released version of which has had an extra parenthesis in a WKT string since forever and which is now caught by GEOS 3.12.0beta. I included `sp` because some packages using `sp` silently assume that `rgdal` oe `rgeos` are available.